js/ai/src/embedder.ts

```diff
@@ -19,14 +19,19 @@ import { Registry } from '@genkit-ai/core/registry';
 import { Document, DocumentData, DocumentDataSchema } from './document.js';
 
 /**
- * Embedding vector.
+ * A batch (array) of embeddings.
  */
-export type Embedding = number[];
+export type EmbeddingBatch = { embedding: number[] }[];
 
 /**
- * A batch (array) of embeddings.
+ * EmbeddingSchema includes the embedding and also metadata so you know
+ * which of multiple embeddings corresponds to which part of a document.
  */
-export type EmbeddingBatch = { embedding: Embedding }[];
+export const EmbeddingSchema = z.object({
+  embedding: z.array(z.number()),
+  embedMetadata: z.record(z.string(), z.unknown()).optional(),
```

Comment:
just `metadata`

Reply:
I started that way. But it gets confusing which metadata is which, so I renamed this one to be clearer.

Reply:
The object is embedding. The metadata field of embedding object is embedding metadata. No need to waste characters.

Reply:
done